packages/middleware-sdk-ec2/package.json

```diff
@@ -23,6 +23,7 @@
     "@aws-sdk/signature-v4": "*",
     "@aws-sdk/types": "*",
     "@aws-sdk/util-format-url": "*",
+    "@aws-sdk/client-ec2": "*",
```

Comment:
This is cause circular dependency of this package and EC2 client. Since we only need interfaces here, you can move it to `devDependencies`. Normally event this is not necessary because we defined the required configs in internal interface `PreviouslyResolved`.

Reply:
switched back to `PreviouslyResolved`